dev/change_version.py

```diff
@@ -0,0 +1,142 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+This script changes versions in every pom.xml and relevant files.
+
+(How to run)
+python change_version <reef_home> <reef_version_for_pom.xml>
+
+(Example)
+python change_version ~/incubator_reef 0.12.0-incubating-SNAPSHOT
+"""
+
+
+import os
+import re
+import sys
+
+"""
+Get list of path for every file in a directory
+"""
+def get_filepaths(directory):
+    file_paths = []  
+
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            file_paths.append(filepath)
+    return file_paths
+
+"""
+Change REEF version to new_version in every pom.xml
+"""
+def change_pom(file, new_version):
+    changed_str = ""
+
+    f = open(file, 'r')
+
+    while True:
+        line = f.readline()
+        if "<version>" in line:
+            break
+        changed_str += line
+
+    r = re.compile('<version>(.*?)</version>')
```

Comment:
@markusweimer Thank you for your comment. At first, I parsed XML and rebuilt it but it ignored comments and made unnecessary changes such as indent or newline. I think this is quite safe because change happens only once for a file.